packages/nodes-base/nodes/Github/Github.node.ts

```diff
@@ -1821,28 +1820,12 @@ export class Github implements INodeType {
 						body.message = this.getNodeParameter('commitMessage', i) as string;
 
 						if (this.getNodeParameter('binaryData', i)) {
-							// Is binary file to upload
-							const item = items[i];
-
-							if (item.binary === undefined) {
-								throw new NodeOperationError(this.getNode(), 'No binary data exists on item!', {
-									itemIndex: i,
-								});
-							}
-
-							const binaryPropertyName = this.getNodeParameter('binaryPropertyName', i);
-
-							if (item.binary[binaryPropertyName] === undefined) {
-								throw new NodeOperationError(
-									this.getNode(),
-									`Item has no binary property called "${binaryPropertyName}"`,
-									{ itemIndex: i },
-								);
-							}
-
 							// Currently internally n8n uses base64 and also Github expects it base64 encoded.
 							// If that ever changes the data has to get converted here.
-							body.content = item.binary[binaryPropertyName].data;
+							const binaryPropertyName = this.getNodeParameter('binaryPropertyName', i);
+							const binaryData = this.helpers.assertBinaryData(i, binaryPropertyName);
+							// TODO: Does this work with filesystem mode
+							body.content = binaryData.data;
```

Comment:
Create Github File does not work with filesystem mode

Reply:
ah yeah. added that note specifically to fix that issue, and then forgot about it.
Should I fix that in this same PR, or should I create a separate PR (since this is actually broken in `master`).

Reply:
Let's fix it in a separate PR than👍. The GitLab one seems to work though with filesystem.